tests/lib/rules/no-shared-component-data.js

```diff
@@ -102,7 +105,18 @@ ruleTester.run('no-shared-component-data', rule, {
           data
         }
       `,
-      parserOptions: { ecmaVersion: 6, sourceType: 'module' }
+      parserOptions
+    },
+    {
+      filename: 'test.vue',
+      code: `
+        export default {
+          data: () => {
+
+          }
+        }
+      `,
```

Comment:
This is sort of correct @mysticatea I think. This plugin doesn't transform `<script></script>` from a `vue` file. It's part of `eslint-vue-parser` responsibilities so here we expect that any code inside `script` tag will be accessible as usual. That's why we write this code directly without extra `script` tag. All we need to know in our rules is the extension of the file to detect if simple `export default` should be treated as component.